docs/guides/build_a_simple_pipeline.md

```diff
@@ -0,0 +1,172 @@
+
+The pipeline starts with the initialization of the image dataset sourced from HuggingFace and it proceeds with the downloading of these carefully selected images. Accomplishing these tasks necessitates the use of a pre-built generic component (HuggingFace dataset loading) and a reusable component (image downloading). 
+
+## Setting up the environment
+To set up your local environment, please refer to our getting started documentation. There, you will find the necessary steps to configure your environment.
+
+## Building the pipeline
+Everything begins with the pipeline definition. Start by creating a 'pipeline.py' file and adding the following code.
+
+```python
+from fondant.pipeline import ComponentOp, Pipeline
+pipeline = Pipeline(
+pipeline_name="creative_commons_pipline", # This is the name of your pipeline
+base_path="./data" # The directory that will be used to store the data
+)
+```
+ 
+
+All you need to initialise a Fondant pipeline are two key parameters:
+- **pipeline_name**: This is a name you can use to reference your pipeline. In this example, we've named it after the creative commons-licensed dataset used in the pipeline.
+- **base_path**: This is the base path that Fondant should use for storing artifacts and data. In our case, it's a local directory path. However, it can also be a path to a remote storage bucket provided by a cloud service. Please note that the directory you reference must exist; if it doesn't, make sure to create it.
+
+## Adding components
+
+Now it's time to incrementally build our pipeline by adding different execution steps. We refer to these steps as `Components`. Components are executable elements of a pipeline that consume and produce dataframes. The components are defined by a component specification. The component specification is a YAML file that outlines the input and output data structures, along with the arguments utilised by the component.
+
+Fondant offers three distinct component types:
+- **Reusable components**: These can be readily used without modification.
+- **Generic components**: They provide the business logic but may require adjustments to the component spec.
+- **Custom components**: The component implementation is user-dependent.
+
+If you want to learn more about components, you can check out the [components documentation](component.md).
+
+### First component to load the dataset
+
+For every pipeline, the initial step is data initialization. In our case, we aim to load the dataset into our pipeline base from HuggingFace. Fortunately, there is already a generic component available called `load_from_hub`.
+This component is categorised as a generic component because the structure of the datasets we load from HuggingFace can vary from one dataset to another. While we can leverage the implemented business logic of the component, we must customise the ComponentSpec. This customization is necessary to inform the component about the specific columns it will produce.
```

Comment:
```suggestion
This component is categorised as a generic component because the structure of the datasets we load from HuggingFace can vary from one dataset to another. While we can leverage the implemented business logic of the component, we must customise the component spec. This customization is necessary to inform the component about the specific columns it will produce.
```